### DISCUSSION TITLE
Welcome announcement, come introduce yourself!

### DISCUSSION THREAD

**Discussion by davidelbert** [marda-alliance/metadata_extractors#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ml-evs** [comment]
Hi everyone, I'm Matthew, one of the leads for this WG and the initial proposer. Although many of us already know each other, I thought it would be a good idea to make some brief intros here and allow people to indicate their expectations of their involvement in this group. 

This WG idea came up in discussions at the [MADICES workshop](https://madices.github.io) I co-organized earlier in the year. I'm primarily interested in this WG as a developer of a data management framework (somewhere between an ELN and a LIMS) for experimental materials chemistry, where it would be super useful to have access to a discoverable repository of high quality parsers for various instrument file formats to enable scalability of more localized data management efforts to various techniques.  

As a WG co-lead, I'm planning to devote around 1 day/week to this project and would like to be involved in the development across all 3 work packages.

Look forward to working with you all (assuming someone actually responds below!)

**Comment by kjappelbaum** [comment]
Hi all, I'm Kevin and was involved with @ml-evs in the organization of the MADICES workshop. 
I try to be involved in the efforts around the [cheminfo ELN](https://cheminfo.github.io/) for which we have been developing many parsers with not super systematic standardization and documentation.

**Reply by PeterKraus** [reply]
I didn't know about cheminfo, looks like a cool project! Are the parsers in it fairly modular (so they could be made stand-alone), or is the integration too deep? What would you like to get out of this WG?

**Comment by PeterKraus** [comment]
Hi everyone, I'm Peter, another one of the leads of this WG. During the last year, in my job at Empa, I've been working on a set of python tools to make the life of my experimental colleagues easier. This includes the instrument automation package [tomato](github.com/dgbowl/tomato), a FAIR data parser [yadg](github.com/dgbowl/yadg), and a post-processing toolkit [dgpost](github.com/dgbowl/dgpost). 

Now I've moved to the TU Berlin, where I'm starting my own group. The project includes both comp. chem. as well as experiments. Having the latter organised in comparable way to what's possible with computational research is why I wanted to contribute to this WG!

**Comment by dwinston** [comment]
Hi all, I'm Donny. I've been involved in systems design and engineering work for the [Materials Project](https://materialsproject.org/), the [Open Databases Integration for Materials Design (OPTIMADE) consortium](https://www.optimade.org/), and MaRDA's [data dictionaries working group](https://github.com/marda-dd). I own and run a small software and data consultancy, [Polyneme LLC](https://polyneme.xyz/), based in New York City, with a focus on scientific data management and stewardship. I've recently branched out into the microbiome science and space physics communities in my work, but I retain a deep interest in advancing the digital capabilities of the materials science community. I'm very much looking forward to working with you all on metadata extraction.

Tooling we may want to consider, either as inspiration or as a target for alignment, is the [Clowder](https://clowderframework.org/) framework's so-called *metadata extractors* ([partial list](https://opensource.ncsa.illinois.edu/confluence/display/CATS/Partial+Extractor+List), [python library to develop said extractors](https://github.com/clowder-framework/pyclowder)).

**Reply by davidelbert** [reply]
Welcome Donny!  Glad you'll be joining us. Clowder is definitely on the list of established options to discuss.  Matthew Evans has some slides speaking to existing things and possible options that he can probably share or perhaps is planning to present in some form at the kick off meeting.  My perspective is that we don't need to replace good solutions as much as create some interoperability so we can all reuse extractors and focus on only having to write new ones if possible.

**Reply by ml-evs** [reply]
Good to see you @dwinston! I'll start a thread for prior art  (edit: just did this https://github.com/marda-alliance/metadata_extractors/discussions/3), as we definitely have a lot to build from --- will definitely reach out to various developers in this area too before the first meeting

**Comment by SteffenBrinckmann** [comment]
Hey,
My name is Steffen Brinckmann (FZ-Jülich, Germany) and work on extractors and ELNs.

There is this working group TAPIR, which I am involved in, that has a similar direction but could benefit from little bump
https://github.com/NFDI4Chem/formaTAPIRest

**Reply by ml-evs** [reply]
This looks great @SteffenBrinckmann (and thanks for adding it to the prior art list). The README example is very similar to what I had in mind here!

**Comment by gmrigna** [comment]
Hi all, I'am Gian-Marco. I am Professor in Materials Science at [UCLouvain](https://uclouvain.be). I am involved in the [ABINIT](https://www.abinit.org) package, the [Materials Project](https://materialsproject.org/), the [NOMAD Center of Excellence](https://www.nomad-coe.eu/), and the [Open Databases Integration for Materials Design (OPTIMADE) consortium](https://www.optimade.org/).

**Comment by eminizer** [comment]
Hi everyone,

I'm Maggie Eminizer, an associate research scientist at the Johns Hopkins Institute for Data Intensive Engineering and Science. I have a PhD in particle physics, but I mostly develop scientific software in Python now. I work closely with David Elbert on a number of his projects in data streaming and automation, and I'm the author of [OpenMSIStream](https://github.com/openmsi/openmsistream), a Python package to simplify standing up streaming ecosystems in lab environments.

I'd love to help out however I can, and I'm excited to learn more from you all : )

**Reply by ml-evs** [reply]
Hi Maggie, great to have you here! I've just added OpenMSIStream to the Prior art list at https://github.com/marda-alliance/metadata_extractors/discussions/3; @davidelbert and I discussed streaming data (and the infrastructure side more generally) as an important use case to cover here.

**Comment by markus1978** [comment]
Hi, I am Markus. I am the lead engineer behind [NOMAD](http://nomad-lab.eu) and [FAIRmat](https://fairmat-nfdi.eu/). NOMAD uses roughly 50 parsers to extract information from various computational materials science codes. We automatically match and run parsers on all files uploaded to NOMAD. There is obviously some overlap with this WG and I am excited to participate.

**Reply by ml-evs** [reply]
Welcome Markus, and thanks for kicking off the discussions in the other threads (as well as at the meeting yesterday)! I have spammed some replies to hopefully continue the ball rolling.

At some point it would be great to have short presentations about particular projects in our monthly meetings. Would you or someone from your team want to give a technical breakdown (parsing or otherwise) of NOMAD/FAIRMat at some point?

**Comment by reetuelzajoseph** [comment]
Hi everyone, I'm Reetu, working at the Karlsruhe Institute of Technology (KIT) in Germany.  I am interested in the working group, because we, at the [Data Exploitation Methods group ](https://github.com/kit-data-manager) are also in the middle of making some metadata extractors and schema mappers and would love to learn from you and contribute with what we have.

**Comment by edan-bainglass** [comment]
Hi everyone, I'm Edan. I work at the Paul Scherrer Institute (PSI), Switzerland, in Dr. Giovanni Pizzi's [Materials Software and Data Group](https://www.psi.ch/en/lms/msd-group) group in part as an [AiiDA](https://www.aiida.net) core developer. I recently got a chance to pick @ml-evs's brain a bit during his PSI visit regarding his MaRDA work. Giovanni is interested, as am I, in attempting to introduce our QE parsers (integrated in the aiida-quantumespresso plugin) into the MaRDA extractor base. I hope to learn more about the extractor system and contribute where I can. Cheers :)

**Comment by BenediktHeinrichs** [comment]
Hello everyone, I am Benedikt Heinrichs and work at the IT Center of the RWTH Aachen University.

In my research work, it was necessary to develop a solution to extract semantic metadata from the research data's content. This work was published as two papers ([10.5220/0010129502270234](https://www.scitepress.org/Link.aspx?doi=10.5220/0010129502270234) and [10.5220/0010654100003064](https://www.scitepress.org/Link.aspx?doi=10.5220/0010654100003064)) and as an [open-source](https://git.rwth-aachen.de/coscine/research/metadataextractor/) [service](https://metadataextractor.otc.coscine.dev/).

A couple of days ago, I was made aware of this initiative and while I am not really a materials scientist, I feel like solutions look these could converge with enough time to a general, multi-domain and extendable metadata extraction solution, which I would be excited about.

**Reply by reetuelzajoseph** [reply]
Welcome Benedikt 👍

**Reply by ml-evs** [reply]
Sounds great, thanks Benedikt!

**Comment by arr19** [comment]
Hi, I'm Anthony from the 2D Crystal Consortium (2DCC) and Materials Characterization Lab at Penn State University. Saw the presentation at the MaRDA virtual meeting and would be happy to port some parsers of experimental data files that we have in Matlab to python to support this.

**Reply by ml-evs** [reply]
Great to have you @arr19! As you probably heard, this WG is wrapping up but we'll try to run some final meetings for tidying up. Could you email me at the address at the bottom of https://ml-evs.science/ and I can make sure you're on the mailing list? 

Would be interested to hear what parsers you have! Do they align with the current registry at all (https://marda-registry.fly.dev/)? I don't know how painful a rewrite would be for you, but we could also discuss adding a `matlab` execution mode (I don't have the matlab expertise to write this myself).

**Reply by PeterKraus** [reply]
If the parsers are compatible with the `octave` flavour of matlab, we could try doing it using `conda`.

**Reply by arr19** [reply]
Several are already listed in the registry, like Quantum Design's .dat format and Panalytical's XRDML. We can discuss others. I should try testing with Octave but those two are simple enough a port would be easy.

**Reply by PeterKraus** [reply]
Perfect, whatever you think is less hassle. We wanted to allow folks to contribute non-python packages, so I just thought I'd ask if you tested it with octave.
